Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -831,23 +832,48 @@ void RenderLayer::rebuildZOrderLists(std::unique_ptr<Vector<RenderLayer*>>& posZ
     }
 }
 
-void RenderLayer::collectLayers(std::unique_ptr<Vector<RenderLayer*>>& positiveZOrderList, std::unique_ptr<Vector<RenderLayer*>>& negativeZOrderList, OptionSet<Compositing>& accumulatedDirtyFlags)
+void RenderLayer::removeSelfAndDescendantsFromCompositor()
 {
-    updateDescendantDependentFlags();
+    if (parent())
+        compositor().layerWillBeRemoved(*parent(), *this);
+    clearBacking();
+
+    for (RenderLayer* child = firstChild(); child; child = child->nextSibling())
+        child->removeSelfAndDescendantsFromCompositor();
+}
+
+void RenderLayer::setWasOmittedFromZOrderTree()
+{
+    if (m_wasOmittedFromZOrderTree)
+        return;
+
+    removeSelfAndDescendantsFromCompositor();
 
+    if (compositor().hasContentCompositingLayers() && parent())
+        parent()->setDescendantsNeedCompositingRequirementsTraversal();
```

Comment:
Is `parent()` enough here, or should we go up to the enclosing stacking context (`paintOrderParent()`)?

Reply:
This is the existing behaviour for `RenderLayer::removeChild`, so hopefully it's sufficient :)